sql/hive-thriftserver/v0.12.0/src/main/scala/org/apache/spark/sql/hive/thriftserver/Shim12.scala

```diff
@@ -184,8 +184,10 @@ private[hive] class SparkExecuteStatementOperation(
   }
 
   def run(): Unit = {
+    val sid = UUID.randomUUID().toString
```

Comment:
No, the listener needs a statement id, not a session id, for updating each statement's state.